smartsim/_core/mli/infrastructure/control/requestdispatcher.py

```diff
@@ -0,0 +1,525 @@
+                    "No request data found.",
+                )
+
+            request_bytes = bytes_list[0]
+            tensor_bytes_list = bytes_list[1:]
+            self._perf_timer.start_timings()
+
+            request = deserialize_message(request_bytes, self._comm_channel_type)
+            if request.input_meta and tensor_bytes_list:
+                request.raw_inputs = tensor_bytes_list
+
+            self._perf_timer.measure_time("deserialize_message")
+            if not self._validate_request(request):
+                return
+
+            self._perf_timer.measure_time("validate_request")
+            self.dispatch(request)
+
+            self._perf_timer.measure_time("dispatch")
+        finally:
+            self.flush_requests()
+            # TODO: implement this
+            # self.remove_queues()
+
+            self._perf_timer.end_timings()
+
+        if self._perf_timer.max_length == 801:
+            self._perf_timer.print_timings(True)
+
+    @property
+    def task_queue(self) -> DragonQueue:
+        """The queue on which batched requests are placed"""
+        return self._outgoing_queue
+
+    def _swap_queue(self, model_key: str) -> None:
+        if self._queue_swap_lock is None:
+            raise SmartSimError("Queues were not locked")
+        with self._queue_swap_lock:
+            for queue in self._queues:
+                if queue.model_key == model_key and not queue.full():
+                    self._active_queues[model_key] = queue
+                    return
+
+            new_queue = BatchQueue(self._batch_timeout, self._batch_size, model_key)
+            self._queues.append(new_queue)
+            self._active_queues[model_key] = new_queue
+            return
+
+    def dispatch(self, request: InferenceRequest) -> None:
+        """Assign a request to a batch queue
+        :param request: the request to place
+        """
+        if request.raw_model is not None:
+            logger.info("Direct inference requested, creating tmp queue")
+            tmp_id = f"_tmp_{str(uuid.uuid4())}"
+            tmp_queue: BatchQueue = BatchQueue(
+                batch_timeout=0, batch_size=1, model_key=tmp_id
+            )
+            self._active_queues[tmp_id] = tmp_queue
+            tmp_queue.put_nowait(request)
+            tmp_queue.make_disposable()
+            return
+
+        if request.model_key:
+            success = False
+            while not success:
+                try:
+                    self._active_queues[request.model_key].put_nowait(request)
+                    success = True
+                except (Full, KeyError):
+                    self._swap_queue(request.model_key)
+
+    def flush_requests(self) -> None:
+        """Get all requests from queues which are ready to be flushed. Place all
+        aviable request batches in the outgoing queue.
+        """
+        for queue in self._queues:
+            if queue.ready and queue.acquire(blocking=False):
```

Comment:
I think this is true, but if the lock can't be aquired will it be called on the next ``_on_iteration``?

Reply:
Yes, an attempt is made at each iteration